### PR TITLE
Use AsciiString, looks less messy

### PR DESCRIPTION
**Pull request by benaadams**

### CODE REVIEW

File: PlatformLevelTechempower/AsciiString.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace PlatformLevelTechempower
+{
+    public struct AsciiString : IEquatable<AsciiString>
+    {
+        private readonly byte[] _data;
+
+        public AsciiString(byte[] bytes) => _data = bytes;
+
+        public AsciiString(string s) => _data = Encoding.UTF8.GetBytes(s);
```
Comment (davidfowl) [inline PlatformLevelTechempower/AsciiString.cs]: We have a faster routine than this. `HttpUtilities. GetAsciiStringNonNullCharacters`

Comment (benaadams) [reply PlatformLevelTechempower/AsciiString.cs]: Haha, it using UTF8 when it shouldn't anyway. Changed to ASCII.

Changed conversion back to string to use `GetAsciiStringNonNullCharacters`

File: PlatformLevelTechempower/AsciiString.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        public bool Equals(AsciiString other) => ReferenceEquals(_data, other._data) || SequenceEqual(_data, other._data);
+        private bool SequenceEqual(byte[] data1, byte[] data2) => new Span<byte>(data1).SequenceEqual(data2);
+
+        public static bool operator ==(AsciiString a, AsciiString b) => a.Equals(b);
+        public static bool operator !=(AsciiString a, AsciiString b) => !a.Equals(b);
+        public override bool Equals(object other) => (other is AsciiString) && Equals((AsciiString)other);
+
+        public override int GetHashCode()
+        {
+            var data = _data;
```
Comment (davidfowl) [inline PlatformLevelTechempower/AsciiString.cs]: Where did you get this impl?

Comment (benaadams) [reply PlatformLevelTechempower/AsciiString.cs]: Copied `string.GetLegacyNonRandomizedHashCode()` dumped all the #if s

https://github.com/dotnet/coreclr/blob/c90a41bb6b9b5c5cc7f8f29a9d06aa2f9d0e51fd/src/mscorlib/src/System/String.Comparison.cs#L982-L1036

Comment (benaadams) [reply PlatformLevelTechempower/AsciiString.cs]: Added comment

**Comment by DamianEdwards** [comment]
This is nice. @benaadams can you rebase and fix conflicts please?

**Comment by benaadams** [comment]
Done

**Pull request merged**